src/main/scala/beam/router/BeamRouter.scala

```diff
@@ -384,6 +384,9 @@ class BeamRouter(
       case None => //No matching id. No need to log since this is more for analysis
     }
 
+  // TODO: availableWorkers is a SET (not sortedSet).
```

Comment:
This seems to be a bug that should be issue-ized?

Reply:
I cannot answer this question

Reply:
@carloscaldas are you suggesting instead to use some data structure which keeps the order, like `TreeSet` so there is expected behavior of getting the first element? Agree, in that case it will be clear how it works, otherwise someone needs to go check that Set is implemented as HashSet which uses, and `Set.head` is actually accessing iterator.

Reply:
@REASY there are 2 possibilities:
- the first element can be any element. in this case we just need to rename the method to make the behavior clear
- it really needs to be the first element. in this case we must make sure the datastructure is sortable

I don't know what is the expected behavior. <-- this is required to propose some solution